project-canvas/src/components/backlog-view/BacklogView.tsx

```diff
@@ -7,81 +7,48 @@ import {
 import { IconChevronLeft } from "@tabler/icons"
-import { Issue } from "project-extender"
+import { Issue, Sprint } from "project-extender"
 import { Column } from "./Column"
 import { resizeDivider } from "./resizeDivider"
 import { onDragEnd } from "./dndHelpers"
 import { useCanvasStore } from "../../lib/Store"
+import { getIssues } from "./issueFetcher"
+import { SprintsColumn } from "./SprintsColumn"
 
 export function BacklogView() {
   const navigate = useNavigate()
   const projectName = useCanvasStore((state) => state.selectedProject?.name)
   const projectKey = useCanvasStore((state) => state.selectedProject?.key)
   const boardIds = useCanvasStore((state) => state.selectedProjectBoardIds)
   const [isLoading, setIsLoading] = useState(true)
-  const [sprints, setSprints] = useState(new Map<string, number>())
-  const updateSprints = (key: string, value: number) => {
-    setSprints((map) => new Map(map.set(key, value)))
-  }
+  const [sprints, setSprints] = useState(new Map<string, Sprint>())
   const [columns, setColumns] = useState(
     new Map<string, { id: string; list: Issue[] }>()
   )
+  const updateSprints = (key: string, value: Sprint) => {
+    setSprints((map) => new Map(map.set(key, value)))
+  }
   const updateColumn = (key: string, value: { id: string; list: Issue[] }) => {
     setColumns((map) => new Map(map.set(key, value)))
   }
 
-  const getIssues = async () => {
-    await Promise.all(
-      boardIds.map(async (boardId) => {
-        const sprintsResponse = await fetch(
-          `${import.meta.env.VITE_EXTENDER}/sprintsByBoardId?boardId=${boardId}`
-        )
-        const sprintsAsArray = await sprintsResponse.json()
-        await Promise.all(
-          sprintsAsArray.map(
-            async (sprint: { sprintId: number; sprintName: string }) => {
-              updateSprints(sprint.sprintName, sprint.sprintId)
-              const issuesForSprintResponse = await fetch(
-                `${
-                  import.meta.env.VITE_EXTENDER
-                }/issuesBySprintAndProject?sprint=${
-                  sprint.sprintId
-                }&project=${projectKey}`
-              )
-              const issuesForSprints = await issuesForSprintResponse.json()
-              updateColumn(sprint.sprintName, {
-                id: sprint.sprintName,
-                list: issuesForSprints,
-              })
-            }
-          )
-        )
-        const backlogIssues = await fetch(
-          `${
-            import.meta.env.VITE_EXTENDER
-          }/backlogIssuesByProjectAndBoard?project=${projectKey}&boardId=${boardId}`
-        )
-        const unassignedPbis = await backlogIssues.json()
-        updateColumn("Backlog", { id: "Backlog", list: unassignedPbis })
-      })
-    )
-    setIsLoading(false)
-  }
-
   useEffect(() => {
-    getIssues()
+    getIssues(projectKey, boardIds, updateColumn, updateSprints, setIsLoading)
   }, [])
 
   useEffect(() => {
     resizeDivider()
   }, [isLoading])
```

Comment:
why does resizeDivider need to be called when isLoading changes? can we remove that?

Reply:
i tried, but it doesn't render properly (it won't let me move the resize divider even though i can see it and click it)